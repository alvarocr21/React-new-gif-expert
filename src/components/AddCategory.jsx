import { useState } from "react";
import PropTypes from "prop-types";

export function AddCategory({ onNewCategory }) {
  //console.log("setCategories", setCategories);
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    //setCategories((categories) => [...categories, inputValue]);
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
