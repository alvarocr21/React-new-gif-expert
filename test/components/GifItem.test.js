import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";
const title = "Goku";
const url = "http://probando/";

describe("Probando componente GifItem.jsx", () => {
  test("debe hacer match con  el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("Muestra los atributos de la imagen", () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug();
    const { alt, src } = screen.getByRole("img");
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test("debe mostrar el texto del titulo", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
