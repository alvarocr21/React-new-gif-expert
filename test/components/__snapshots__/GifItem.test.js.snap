// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente GifItem.jsx debe hacer match con  el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="http://probando/"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
